src/registry/BaseRegistry.ts

```diff
@@ -59,11 +58,44 @@ export abstract class BaseRegistry implements IRegistry {
     return 'deployments/warp_routes';
   }
 
-  protected getWarpRouteCoreConfigPath(config: WarpCoreConfig, options?: AddWarpRouteOptions) {
-    return `${this.getWarpRoutesPath()}/${warpRouteConfigToId(
-      config,
-      options?.symbol,
-    )}-config.yaml`;
+  /**
+   * Gets a warp route ID from a warp route config.
+   * This uses the first symbol in the list. Situations where a config contains multiple
+   * symbols are not officially supported yet.
+   */
+  static warpRouteConfigToId(
+    config: WarpCoreConfig,
+    options?: AddWarpRouteConfigOptions,
+  ): WarpRouteId {
+    let warpRouteId;
+    if (options && 'warpRouteId' in options) {
+      warpRouteId = options.warpRouteId;
+    } else {
+      if (!config?.tokens?.length) throw new Error('Cannot generate ID for empty warp config');
```

Comment:
The logic here should be the same as the existing warpRouteConfigToId() found in the utils